core/src/main/java/jenkins/telemetry/impl/java11/MissingClassTelemetry.java

```diff
@@ -0,0 +1,283 @@
+     * pieces of information we want to gather are the places where the {@link ClassNotFoundException} or the
+     * {@link NoClassDefFoundError} errors happens, rather than the class itself.
+     * @param events events collected in this telemetry window.
+     * @return the events formatted in a map.
+     */
+    @Nonnull
+    private JSONArray formatEvents(@Nonnull ConcurrentHashMap<List<StackTraceElement>, MissingClassEvent> events) {
+        JSONArray jsonEvents = new JSONArray();
+
+        events.forEach((stackTrace, event) -> {
+            JSONObject eventObject = new JSONObject();
+            eventObject.put("className", event.getClassName());
+            eventObject.put("class", event.getClassName());
+            eventObject.put("time", event.getTime());
+            eventObject.put("occurrences", Long.toString(event.getOccurrences()));
+            eventObject.put("stacktrace", event.getStackTrace());
+
+            jsonEvents.add(eventObject);
+        });
+
+        return jsonEvents;
+    }
+
+    /**
+     * The current time in the same way as other telemetry implementations.
+     * @return the UTC time formatted with the pattern yyyy-MM-dd'T'HH:mm'Z'
+     */
+    @Nonnull
+    static String clientDateString() {
+        TimeZone tz = TimeZone.getTimeZone("UTC");
+        DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'");
+        df.setTimeZone(tz); // strip timezone
+        return df.format(new Date());
+    }
+
+    /**
+     * Store the exception if it's from a split package of Java. This method report this exception directly, it doesn't
+     * look into the causes or suppressed exceptions of the exception specified. This method tends to be used in the
+     * ClassLoader directly. Outside the class loaders is best to use {@link #reportExceptionInside(Throwable)}
+     * @param name the name of the class
+     * @param e the throwable to report if needed
+     */
+    public static void reportException(@Nonnull String name, @Nonnull Throwable e) {
+        //ClassDefFoundError uses / instead of .
+        name = name.replace('/', '.').trim();
+
+        if (isFromMovedPackage(name)) {
+            events.put(name, e);
+
+            if (LOGGER.isLoggable(Level.FINE)) LOGGER.log(Level.FINE, "Added a missed class for Java 11 telemetry. Class: " + name, e);
+        }
+    }
+
+    /**
+     * Store the exception extracting the class name from the message of the throwable specified. This method report
+     * this exception directly, it doesn't look into the causes or suppressed exceptions of the exception specified.
+     * This method tends to be used in the ClassLoader directly. Outside the class loaders is best to use
+     * {@link #reportExceptionInside(Throwable)}
+     * @param e the exception to report if needed
+     */
+    public static void reportException(@Nonnull Throwable e) {
+        String name = e.getMessage();
+
+        if (name == null || name.length() == 0) {
```

Comment:
if `name` is "   " would execute the else clause
```suggestion
        if (String.isEmpty(name)) {
```

Reply:
I've realised you should add also the `import` clause :sweat_smile:

Reply:
I suppose you meant `StringUtils.isEmpty`. But with this method, the outcome is the same, see the code: https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/StringUtils.java#L212

I've added a commit to take into account things like " " or  "       ". Unlikely anyway.

Reply:
You're right, I meant StringUtils. :man_facepalming: